Add image processors refactor to v5 migration guide (#45556)
huggingface/transformers: MIGRATION_GUIDE_V5.md

- Minor change: `XXXFastImageProcessorKwargs` is removed in favor of `XXXImageProcessorKwargs` which will be shared between fast and slow processors (https://github.com/huggingface/transformers/pull/40931)

> ### Image Processors

Reviewer: ```suggestion
### Image processors
```

> The old slow/fast dual-file design — a PIL-based `image_processing_<model>.py` paired with a torchvision-based `image_processing_<model>_fast.py` — has been replaced with a named-backend architecture:

Reviewer: ```suggestion
The old slow/fast dual-file design has been replaced with a named-backend architecture. Each model previously had a PIL-based `image_processing_<model>.py` and a torchvision-based `image_processing_<model>_fast.py`. The new layout is:
```

- `image_processing_<model>.py` → **torchvision** backend (default; was previously `FooImageProcessorFast`)
- `image_processing_pil_<model>.py` → **PIL** backend (was previously `FooImageProcessor`)

> Processor classes now inherit from `TorchvisionBackend` or `PilBackend` (defined in `image_processing_backends.py`), which provide ready-made implementations of all standard operations (`resize`, `rescale`, `normalize`, `center_crop`, `pad`) and a default `_preprocess` pipeline. `BaseImageProcessor` (in `image_processing_utils`) handles the shared preprocessing boilerplate — kwargs validation, default-filling from class attributes, and input preparation — so model-specific processors contain only what is genuinely unique to the model. Most processors now simply inherit from a backend and declare class-attribute defaults; only processors with custom logic (e.g. patch tiling) need to override `_preprocess`.

Reviewer: ```suggestion
Processor classes now inherit from `TorchvisionBackend` or `PilBackend` (defined in `image_processing_backends.py`), which provide ready-made implementations of all standard operations (`resize`, `rescale`, `normalize`, `center_crop`, `pad`) and a default `_preprocess` pipeline. `BaseImageProcessor` (in `image_processing_utils`) handles shared preprocessing boilerplate: kwargs validation, default-filling from class attributes, and input preparation. Model-specific processors contain only what is unique to the model. Most processors inherit from a backend and declare class-attribute defaults. Only those with custom logic (e.g. patch tiling) need to override `_preprocess`.
```
